consensus/state_processing/src/per_block_processing.rs

```diff
@@ -547,11 +553,26 @@ pub fn process_withdrawals<'payload, T: EthSpec, Payload: AbstractExecPayload<T>
                 )?;
             }
 
+            // Update the next withdrawal index if this block contained withdrawals
             if let Some(latest_withdrawal) = expected_withdrawals.last() {
                 *state.next_withdrawal_index_mut()? = latest_withdrawal.index.safe_add(1)?;
-                let next_validator_index = latest_withdrawal
-                    .validator_index
-                    .safe_add(1)?
+
```

Comment:
this warped my brain too much I'm embarrassed to say.. but it's equivalent to the spec so long as `MAX_WITHDRAWALS_PER_PAYLOAD != 0`

Reply:
Hmm I wonder if that's a safe assumption to make or if we should rewrite it to look more like the spec.

I wrote it like this to avoid bounds-checking multiple times, but that's probably insignificant and could be optimised out by the compiler anyway

Reply:
Yeah I figure that was why. There are staking schemes where withdrawals would never be allowed (this is one way to mitigate long-range attacks), and so another protocol forking us and setting this to zero might be a way to achieve this.. idk

Reply:
actually the spec is also ambiguous in the case that `MAX_WITHDRAWALS_PER_PAYLOAD = 0`:
```python
    if len(expected_withdrawals) == MAX_WITHDRAWALS_PER_PAYLOAD:
        # the line below implicitly assumes `MAX_WITHDRAWALS_PER_PAYLOAD != 0`
        next_validator_index = ValidatorIndex((expected_withdrawals[-1].validator_index + 1) % len(state.validators))
        state.next_withdrawal_validator_index = next_validator_index
```

Reply:
so tldr; it's probably fine as it is

Reply:
I think the spec is even explicitly disallowing this, if `expected_withdrawals == []` then `expected_withdrawals[-1]` will panic.

Good thing we've analysed it so thoroughly tho